cogboard-app/src/main/kotlin/com/cognifide/cogboard/widget/type/logviewer/LogViewerWidget.kt

```diff
@@ -0,0 +1,118 @@
+package com.cognifide.cogboard.widget.type.logviewer
+
+import com.cognifide.cogboard.CogboardConstants.Props
+import com.cognifide.cogboard.config.service.BoardsConfigService
+import com.cognifide.cogboard.widget.BaseWidget
+import com.cognifide.cogboard.widget.Widget
+import com.cognifide.cogboard.widget.connectionStrategy.ConnectionStrategy
+import com.cognifide.cogboard.widget.connectionStrategy.ConnectionStrategyFactory
+import com.cognifide.cogboard.widget.type.logviewer.logparser.LogParserStrategy
+import com.cognifide.cogboard.widget.type.logviewer.logparser.LogParserStrategyFactory
+import io.vertx.core.Vertx
+import io.vertx.core.eventbus.Message
+import io.vertx.core.eventbus.MessageConsumer
+import io.vertx.core.json.JsonObject
+
+class LogViewerWidget(
+    vertx: Vertx,
+    config: JsonObject,
+    serv: BoardsConfigService
+) : BaseWidget(vertx, config, serv) {
+    private val address = config.endpointProp(Props.URL)
+    private var consumer: MessageConsumer<*>? = null
+    private val connectionStrategy: ConnectionStrategy = determineConnectionStrategy()
+    private val logParsingStrategy: LogParserStrategy = determineLogParsingStrategy()
+
+    override fun start(): Widget {
+        consumer = connectionStrategy.getConsumer(eventBusAddress)
+        consumer!!.handler {
+            handleResponse(it)
+        }
+        return super.start()
+    }
+
+    override fun stop(): Widget {
+        consumer?.unregister()
+        return super.stop()
+    }
+
+    // private val logs = createTemplateLogs(20)
+    override fun updateState() {
+        if (address.isNotBlank()) {
+            connectionStrategy.sendRequest(address, config)
+        } else {
+            sendConfigurationError("Endpoint URL is blank")
+        }
+
+        // send(JsonObject().put("logs", logs))
+    }
+
+    // private fun createTemplateLogs(n: Int): JsonArray {
```

Comment:
co to za komentarze? wyrzućmy, żeby nie śmiecić kodu

Reply:
Done